### ISSUE TITLE
Logitech wrapper does not pass through to Logitech devices

### ISSUE THREAD

**Issue by KaHonu** [antonpup/Aurora#1726]
### Expected Behaviour
Logitech devices should still receive RGB commands while the Logitech wrapper is in use

### Actual Behaviour
LGS is installed to control Logitech devices. When the wrapper is installed and replaces the official Logitech dll files Logitech devices no longer respond to true Logitech SDK calls.

### Reproduction steps
Install LGS, confirm lighting is working as expected on Logitech devices. Install Aurora, confirm lighting is still working as expected on Logitech devices. Install Logitech wrapper and observe lighting no longer works as expected with Logitech Devices (while Aurora is still running).

### Frequency
Every time

### Aurora Version:
v0.7.0rc2

### Previously an Issue?
Unknown, but most likely

### Additional Notes

The program I used to troubleshoot this issue is located here: https://apps.magicmau.nl/EliteG19s-v2/

It's a Logitech "ARX control" applet that also controls RGB lighting on Logitech devices. It could probably be best described as a "generic app" that controls RGB lighting using the Logitech SDK.

I'm not sure if this should be considered a bug or enhancement.

I don't have any Razer devices to test with, but this program (A Razer to Corsair wrapper) does pass through to Razer devices:

https://github.com/MartB/RETC

Edit for clarification: I am using a combination of Corsair and Logitech devices. Without the wrapper only Logitech devices light up as expected. With the wrapper installed no devices light up. If I install only Corsair devices the wrapper works as expected.

**Comment by diogotr7** [comment]
As far as I know retc does the same, also needs a background program to just relay the information just like aurora. I don't think this is worth fixing. What might be worth implementing is an easier way to revert the changes made to both the dlls and the registry.

**Comment by KaHonu** [comment]
If I revert changes that would put everything back in a state where only the Logitech devices display RGB activity for games with native Logitech support (versus no devices at all with the wrapper in place - which tells me it might be broken for Corsair). I thought the point of Aurora was to have all connected RGB devices working in sync regardless of vendor. In this case I want to put the wrapper in place so that my Corsair keyboard and Logitech mouse / powerplay pad / G13 all display the same RGB events.

Not sure if this helps, but if I have only Corsair devices connected the Logitech wrapper works as expected.

**Comment by diogotr7** [comment]
If you have the wrapper applied and working correctly, it doesn't matter which devices you have connected to aurora, the lighting will work with everything (except a few layout related exceptions, which you can fix with the key cloning in the Wrapper Lights layer).

>Not sure if this helps, but if I have only Corsair devices connected the Logitech wrapper works as expected.

This doesn't really make sense, why would another device brand mess with the wrapper when connected to Aurora?

**Comment by diogotr7** [comment]
As a clarification, you should be able to have different keyboards from different brands (INCLUDING logitech) connected to Aurora and all of them should display the same lighting without issues.

**Comment by KaHonu** [comment]
That's what I'm trying to accomplish here. I'm using a program that has native Logitech RGB support. Devices are a mix of Logitech and Corsair. Without the Logitech wrapper installed only Logitech devices respond to RGB calls (as expected). When I install the wrapper none of the devices respond to RGB calls. If I only have Corsair products connected and install the wrapper everything lights up as expected. Something is broken with the 'mixed device' scenario. I *think* I may have had the Corsair keyboard working at some point with the wrapper and Logitech devices but I don't quite remember. What I can say for sure is that when the wrapper is in use Logitech devices no longer respond to RGB calls.

If it helps I can test any scenario you request. Devices I have are Logitech g910, G903, G13, Powerplay Mat; Corsair Strafe RGB, Dark Core SE. Powerplay and G13 both respond like mouse devices when used in Aurora.

**Comment by KaHonu** [comment]
Quick update. Only Corsair Strafe RGB and G903 connected. I wiped out iCUE, LGS, and Aurora settings after uninstalling each. I reinstalled iCUE and Aurora. Strafe RGB was set as the preferred keyboard in Aurora, no mouse was selected. I ran the app noted above and GTA V -- no lighting effects (as expected). I installed the Logitech wrapper and tried each game again. Still no lighting. Could there be a problem with the Logitech wrapper and Corsair devices?

**Comment by KaHonu** [comment]
Here's another update. If these are getting excessive let me know and I'll delete / edit existing ones.

I uninstalled / cleaned up again and installed LGS 9.02.65, iCUE 3.20.80, and Aurora v0.6.4b.

On the new install Aurora recognized the presence of Logitech and Corsair software (I got the 'first time' prompts for both). This did not happen with the v0.7.x releases -- I only got the Logitech first time prompt.

The Logitech wrapper now sends RGB data to the Corsair device as expected. My Logitech mouse still does not work with the wrapper. There appear to be two issues:

1) v0.7.x Logitech wrapper does not work with Corsair devices; Aurora may also not be accurately identifying the presence of Corsair software or devices

2) Logitech wrapper does not pass RGB info on to Logitech devices.
